src/zcash/address/zip32.h

```diff
@@ -118,13 +329,14 @@ struct SaplingExtendedSpendingKey {
     }
 
     static SaplingExtendedSpendingKey Master(const HDSeed& seed);
+    static std::pair<SaplingExtendedSpendingKey, CKeyMetadata> ForAccount(const HDSeed& seed, uint32_t bip44CoinType, uint32_t accountId);
+    static std::pair<SaplingExtendedSpendingKey, CKeyMetadata> Legacy(const HDSeed& seed, uint32_t bip44CoinType, uint32_t addressIndex);
+
 
     SaplingExtendedSpendingKey Derive(uint32_t i) const;
 
     SaplingExtendedFullViewingKey ToXFVK() const;
 
-    libzcash::SaplingPaymentAddress DefaultAddress() const;
```

Comment:
I started out by hoping to make it possible to remove 'DefaultAddress' entirely, with the thought that the default address for a Sapling key might not have a consistent independent meaning in the context of unified addresses: for example, if the minimum valid diversifier index for Sapling were to produce an error in the transparent key tree derivation, that would not be the same as the default address in the context of the UIVK. However, the default address was used in so many places that I ended up deciding that to be too big of a change. I'm now considering whether it might actually be worth it to explicitly use `.FindAddress(diversifier_index_t(0)).first` instead though.

Reply:
I'm going to leave this as-is.